### PR TITLE
Fix version test

### PR DESCRIPTION
**Pull request by ali-tny**
Before, development version tags (like 0.10.4-dev-optional-import) didn't match the test regex,
leading to failure. This commit changes the regex to also match development tags.

### CODE REVIEW

**Comment by ig248** [comment]
aside: we should consider using [versioneer](https://github.com/warner/python-versioneer)

**Comment by ig248** [comment]
@ali-tny thanks for fixing this. Come to think of it, I have introduced the issue when used a "non-semver" tag `0.10.4-dev-optional-import` - I think before trying to catch all edge cases, I would rather avoid using custom tags on master.

**Comment by ali-tny** [comment]
yep sounds good will change

File: tests/test_version.py [hunk 1]
```diff
@@ -5,5 +5,25 @@
 
 def test_version():
     ver = get_git_version(abbrev=7)
-    VER_RE = r"^(\d+.\d+.\d+)(-\d+)*(-g\S{7})?(-dirty)?$"
+    VER_RE = re.compile(
```
Comment (ig248) [inline tests/test_version.py]: @ali-tny could you add a comment/link to `semver` project in case we need to revisit this ever

Comment (ali-tny) [reply tests/test_version.py]: 👍
